react-components/src/hooks/useSearchMappedEquipmentFDM.tsx

```diff
@@ -0,0 +1,312 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { type CogniteClient } from '@cognite/sdk/dist/src';
+import { useMemo } from 'react';
+import {
+  type EdgeItem,
+  type NodeItem,
+  FdmSDK,
+  type Source,
+  type ViewItem,
+  type Query
+} from '../utilities/FdmSDK';
+import { useSDK } from '../components/RevealContainer/SDKProvider';
+import { type UseQueryResult, useQuery } from '@tanstack/react-query';
+import { SYSTEM_3D_EDGE_SOURCE, SYSTEM_3D_NODE_TYPE } from '../utilities/globalDataModels';
+import { type AddModelOptions } from '@cognite/reveal';
+import { isEqual } from 'lodash';
+
+export type SeachResultsWithView = { view: Source; instances: NodeItem[] };
+
+export const useSearchMappedEquipmentFDM = (
+  query: string,
+  spacesToSearch: string[],
+  models: AddModelOptions[],
+  limit: number = 100,
+  userSdk?: CogniteClient
+): UseQueryResult<SeachResultsWithView[]> => {
+  if (limit > 1000) {
+    throw new Error('Limit cannot be greater than 1000');
+  }
+
+  const sdk = useSDK(userSdk);
+
+  const fdmSdk = useMemo(() => new FdmSDK(sdk), [sdk]);
+
+  const viewsToSearchPromise: Promise<Source[]> = useMemo(
+    async () => await getViewsToSearch(fdmSdk, spacesToSearch),
+    [spacesToSearch, fdmSdk]
+  );
+
+  return useQuery(
+    ['reveal', 'react-components', 'search-mapped-fdm', query, models, viewsToSearchPromise],
+    async () => {
+      const viewsToSearch = await viewsToSearchPromise;
+
+      if (query === '') {
+        const result = await fdmSdk.queryNodesAndEdges(
+          createMappedEquipmentQuery(models, viewsToSearch, limit)
+        );
+
+        const transformedResults = convertQueryNodeItemsToSearchResultsWithViews(
+          result.items.mapped_nodes
+        );
+
+        return transformedResults;
+      }
+
+      const searchResults: SeachResultsWithView[] = [];
+
+      for (const view of viewsToSearch) {
+        const result = await fdmSdk.searchInstances(view, query, 'node', limit);
+        searchResults.push({ view, instances: result.instances });
+      }
+
+      const filteredSearchResults = filterSearchResultsByMappedTo3DModels(
+        fdmSdk,
+        searchResults,
+        models
+      );
+
+      return await filteredSearchResults;
+    },
+    { staleTime: Infinity }
+  );
+};
+
+export const useAllMappedEquipmentFDM = (
+  models: AddModelOptions[],
+  spacesToSearch: string[],
+  userSdk?: CogniteClient
+): UseQueryResult<NodeItem[]> => {
```

Comment:
I think we should remove the empty "view/version" objects in this result. I don't think they add any value when returned to the user

Reply:
Ok, removed empty properties for both hooks here.